### PR TITLE
fix(db): emit change on order-only reorder in ordered live queries

### PR DESCRIPTION
**Pull request by v-anton**
## 🎯 Changes

A live query with `orderBy` does not emit a change when a row's position changes but its projected value does not. For example:

```ts
q.from({ s: source })
 .orderBy(({ s }) => s.value)
 .select(({ s }) => ({ id: s.id }))   // sort field not in projection
```

When a row's `value` changes so its position moves, the projected value (`{ id }`) is byte-identical, so the collection's value-diff (`deepEquals` in `state.ts`) suppresses the update. The live query's `SortedMap` re-sorts internally, but `subscribeChanges` emits nothing — so `useLiveQuery` keeps rendering the **stale order**. Any projection that omits the changing `orderBy` field hits this (it is not specific to id-only).

### Fix

The `orderBy` operator already streams the move as a retract (old value + old `orderByIndex`) plus an insert (new value + new index). We capture the retracted side in `accumulateChanges` and, after each flush, emit an `update` directly **only** when the index changed **and** the value is unchanged — the exact case the value-diff swallows. This mirrors the existing includes-materialization direct-emit pattern and leaves generic collection code untouched.

- Zero cost for collections without `orderBy` (the gate short-circuits on `undefined` index).
- No double-emit: the gate (value unchanged) is mutually exclusive with `state.ts`'s emit (value changed).
- Only adds emits; never removes or alters existing ones.

## ✅ Checklist

- [x] Tested locally with `pnpm test`.

New tests: `packages/db/tests/live-query-orderby-reorder.test.ts` (id-only reorder emits; exactly-one-emit when sort field projected; no spurious emit; `limit` window) and `packages/react-db/tests/useLiveQuery-orderby-reorder.test.tsx` (`useLiveQuery` renders correct order). Full `@tanstack/db` (2391) and `@tanstack/react-db` (95) suites pass.

## 🚀 Release Impact

- [x] Affects published code; changeset included (`@tanstack/db` patch).

## Bug Fixes
* Fixed live queries with `orderBy` to correctly detect and publish “order-only” reorders, even when item projected values remain unchanged.
* Prevented duplicate change notifications during reorders, including cases where the sort field is included or excluded from the selected projection.

## Tests
* Expanded live-query `orderBy` reorder coverage (projection behavior, `limit`/`offset` edge cases, single-item, and consecutive reorders).
* Added React `useLiveQuery` coverage to verify UI ordering updates after reorder writes even when the sort field isn’t selected.

### CODE REVIEW

**Comment by v-anton** [comment]
### Relationship to other open `orderBy` issues

While investigating I checked the nearby open issues in the ordered-query area. This fix is **related but distinct** from both:

- **#1489** (chained live-query over `leftJoin` misses same-key value updates, root-caused to `TopKState.processElement`): I ran the issue's exact reproduction against current `main` and it **passes with and without this PR** — it appears to have been resolved upstream since the reporter's `@tanstack/db@0.6.5`. Not addressed here because it no longer reproduces; #1489 may be closeable.
- **#1444** (`includes` ignores `orderBy` after an optimistic child update): I reproduced it at the `db` layer and it **still fails on `main`, with and without this PR**. It lives in the `includes`/subquery path (which has its own emit), whereas this PR fixes the top-level ordered result. Distinct bug, not fixed here.

This PR specifically fixes the case where a top-level ordered live query's projection omits the changing `orderBy` field, so a reorder is suppressed by the collection's value-diff (`state.ts` `deepEquals`) and never reaches subscribers.

File: packages/db/tests/live-query-orderby-reorder.test.ts [hunk 1]
```diff
@@ -0,0 +1,155 @@
+import { describe, expect, it } from 'vitest'
+import { createCollection, createLiveQueryCollection } from '../src/index'
+
+const flush = () => new Promise((r) => setTimeout(r, 20))
```
[the remark this answers is not in the document]
Comment (v-anton) [reply packages/db/tests/live-query-orderby-reorder.test.ts]: Addressed in 18f158d5 — the live query processes synchronously on commit, so I dropped the fixed `setTimeout` flushes entirely and the tests now assert directly after each write (fully deterministic, no sleeps).

**Comment by v-anton** [comment]
Thanks @coderabbitai — addressed the test feedback in `18f158d5`:

- **Flaky fixed `setTimeout` flushes** → removed. The live query processes synchronously on commit, so the tests now assert directly after each write (no sleeps, deterministic).
- **`limit`/`offset` boundary cases** → added `limit(0)` and offset-beyond-length tests asserting the empty window stays empty and silent across a reorder.
- **`any` in the test** → typed the sync helper (`SyncFns` via `SyncConfig`) and projections (`idsOf`).

On the **`(config.collection as any)._changes.emitEvents(...)`** nitpick: I've left it as-is intentionally. It mirrors the existing includes-materialization path in the same file (`collection-config-builder.ts`, the `inlineReEmitKeys` block), which uses the identical `(collection as any)._changes` pattern to force-emit. Introducing a new public `Collection.emitInternalChanges(...)` delegate to satisfy this one call would add public API surface and leave that established precedent inconsistent — better handled as a separate, focused refactor of both sites if the maintainers want it. Happy to do that follow-up on request.

**Comment by v-anton** [comment]
Added the two requested corner cases in `02a6863f`: a **single-element collection** (sort-field change can't reorder → stays silent) and **back-to-back opposite reorders** (move to front, then push back → asserts correct final order and that each reorder is observed). Suite now 8 cases; full `@tanstack/db` suite green.

**Comment by v-anton** [comment]
Good call — tightened in the latest commit: the back-to-back test now asserts `toBe(1)` after the first reorder and `toBe(2)` after the second, so it fails on any duplicate emission (the exact property this PR guards). Still green.

**Comment by KyleAMathews** [comment]
Closing as superseded by current main. The merged order-only publication suite retains the invariant across direct, limited, included, and deeply nested ordered queries, including exactly-once publication. The remaining ordered-window defects stay tracked by #1555, #1556, and RFC #1657.
